python/GafferArnoldUI/ArnoldOptionsUI.py

```diff
@@ -1089,6 +1091,25 @@ def __gpuSummary( plug ) :
 
 		],
 
+		"populateGPUCache" : [
+
+			"description",
+			"""
+			Pre-populates the GPU program cache with commonly used scene configurations,
+			potentially reducing GPU render startup time. Caches are unique per user and
+			Arnold version. Caches persist between Gaffer sessions and reboots. Some
+			features may still need to be cached when they are first used, causing a
+			delay in GPU render startup.
+			""",
+
+			"nodule:type", "",
+			"label", "Populate GPU Cache",
+			"layout:visibilityActivator", "deviceIsGPU",
```

Comment:
What I was hoping for in ArnoldOptionsUI was a way of populating the cache _before_ a GPU render happens. When switching from CPU to GPU in a live render (which I think may be common for the viewer), it's already too late if the button appears only after the switch has happened. 

I thought the ideal workflow would be that there isn't any button, but when you pick GPU from the dropdown, we automatically start populating the cache if necessary. But I think we can only do that by butchering PresetsPlugValueWidget a bit. And since we can't actually determine if we need to populate the cache or not, and repeated population still seems to take significant time, I guess this isn't a good idea anyway.

Maybe we just drop this commit? Or use a custom widget rather than a plug and remove the activator?

Reply:
If we do keep it, let's move the button inside the "GPU" section.

Reply:
I think it needs repopulating for every Arnold version change and every driver version change, so not completely infrequently. But in the absence of a better plan, let's drop it for now...

Reply:
>  let's move the button inside the "GPU" section

It will be moot if we drop this commit, but for my own education, I'm not sure how to do that. I tried putting it in `GPU`, `Settings.GPU`, `Settings.Options.GPU` and maybe a few others but the closest I got to success was getting a second `GPU` section along with the existing one.

I'm guessing the `CompoundDataPlug` is handling sections on its own, which complicates this effort? Is there a way to bash it in manually?

Reply:
Ahhhh, ok, very good to know. And commit has been dropped now.